tests/test_sft_trainer.py

```diff
@@ -238,8 +221,27 @@ def test_resume_training_from_checkpoint_with_flag_checkpoint_path():
         assert final_trainer_state["total_flos"] == init_trainer_state["total_flos"]
 
 
-def _get_latest_checkpoint_trainer_state(dir_path, checkpoint_index=-1):
+def _get_latest_checkpoint_trainer_state(dir_path: str, checkpoint_index: int = -1):
+    """
+    Get the trainer state from the specified checkpoint directory.
+    This function gets the latest or specific checkpoint based on the
+    provided checkpoint_index from the checkpoint directory, and loads
+    the `trainer_state.json` file from that checkpoint. The trainer
+    state is returned along with the path to the checkpoint.
```

Comment:
nit: kinda repetitive so makes more succinct
```suggestion
    Get the trainer state from the latest or specified checkpoint directory.
    The trainer state is returned along with the path to the checkpoint.
```

Reply:
Thank you!